### ISSUE TITLE
Streamline plot callback may be transposed

### ISSUE THREAD

**Issue by yt-fido** [yt-project/yt#1009]
Originally reported by: **Matt Turk (Bitbucket: [MatthewTurk](https://bitbucket.org/MatthewTurk), GitHub: [MatthewTurk](https://github.com/MatthewTurk))**

----------------------------------------

In debugging the streamline plot callback, I've run into something odd.  I'm hoping @jwise77 , who wrote it originally, might be able to chime in.

It's not clear to me that we are doing "the right thing" since we have an image plot which is using `origin='lower'` and a streamline plot that is not.  Additionally, the x, y ordering requires us *not* to transpose the resultant image buffers to get the right shape of our array, if we have a non-unitary input buffer.

In short, I think the inputs may be transposed but shouldn't be, and I'm not sure the streamplot is right.

----------------------------------------
- Bitbucket: https://bitbucket.org/yt_analysis/yt/issue/1009
